worlds/pokemon_rb/rom.py

```diff
@@ -1,21 +1,65 @@
 import os
-import hashlib
-import Utils
-import bsdiff4
 import pkgutil
-from worlds.Files import APDeltaPatch
-from .text import encode_text
+import typing
+
+from settings import get_settings
+import Utils
+from worlds.Files import APProcedurePatch, APTokenMixin, APTokenTypes
+
+from . import poke_data
 from .items import item_table
+from .text import encode_text
 from .pokemon import set_mon_palettes
+from .regions import PokemonRBWarp, map_ids, town_map_coords
 from .rock_tunnel import randomize_rock_tunnel
 from .rom_addresses import rom_addresses
-from .regions import PokemonRBWarp, map_ids, town_map_coords
-from . import poke_data
 
+if typing.TYPE_CHECKING:
+    from . import PokemonRedBlueWorld
+
+
+class PokemonRedProcedurePatch(APProcedurePatch, APTokenMixin):
+    game = "Pokemon Red and Blue"
+    hash = "3d45c1ee9abd5738df46d2bdda8b57dc"
+    patch_file_ending = ".apred"
+    result_file_ending = ".gb"
+
+    procedure = [
+        ("apply_bsdiff4", ["base_patch.bsdiff4"]),
+        ("apply_tokens", ["token_data.bin"]),
+    ]
+
+    @classmethod
+    def get_source_data(cls) -> bytes:
+        with open(get_settings().pokemon_rb_options.red_rom_file, "rb") as infile:
```

Comment:
ideally lazy load the world and use this
```suggestion
        with open(PokemonRedBlueWorld.settings.red_rom_file, "rb") as infile:
```

Reply:
I expected VS Code to maybe pick up on the typing, but it doesn't seem to. Still works either way.